service/controller/clusterapi/v19/controllercontext/status.go

```diff
@@ -2,4 +2,10 @@ package controllercontext
 
 type ContextStatus struct {
 	Versions map[string]string
+	Worker   ContextStatusWorker
+}
+
+type ContextStatusWorker struct {
+	Nodes int
+	Ready int
```

Comment:
The `workercount` resource puts the worker counts combined for all machine deployments into the controller context.

Reply:
I think it would make more sense to have something like:
```
type ContextStatus struct {
	Versions map[string]string
	Workers   map[string]ContextStatusNodePoolWorkers
}

type ContextStatusNodePoolWorkers struct {
	Nodes int
	Ready int
}
```

Then the key in this map would be the name of each nodepool. For the upgrade logic in the future (no matter which route we go) it will be beneficial to know to which nodepools which number of workers is ready.

Reply:
I only saw your message when I merged. We can adapt this. Makes sense.